DoSomething/gambit-conversations
# Import Message Broadcast TLC - part 1

Splits the `lib/middleware/import-message/broadcast` into separate `broadcast-get` and `parse-broadcast` files, with some cleanup to Contentful.

* Adds a `contentfulId` function to  `lib/contentfulQueryBuilder` to fetch a Broadcast by its Contentful entry ID instead of the `broadcastId` field value -- fixes #116 

* Moves throwing a 404 if broadcast not found into `contentful. fetchSingleEntry`, deprecating `helpers.getBroadcast`. A `POST /import-message` with a non-existing `broadcastId` will now throw a 404 instead of a 500 `Internal Server Error` - fixes #118 

* Adds the `broadcastId` as a New Relic parameter - refs #214 

To test:
* `POST /import-message` with both valid and invalid broadcast ID's to verify expected API response.

For Part 2:

* Add New Relic params and tests for `parse-broadcast` middleware
* Complete the TODO's of moving the Broadcast parsing functions (e.g. `getTopic(broadcastObject)` into `lib/contentful`
